src/bootstrap/src/core/builder/tests.rs

```diff
@@ -2065,6 +2065,150 @@ mod snapshot {
         [doc] rustc 1 <host> -> reference (book) 2 <host>
         ");
     }
+
+    #[test]
+    fn clippy_ci() {
+        let ctx = TestCtx::new();
+        insta::assert_snapshot!(
+            ctx.config("clippy")
+                .path("ci")
+                .render_steps(), @r"
+        [clippy] rustc 0 <host> -> bootstrap 1 <host>
+        [build] llvm <host>
+        [build] rustc 0 <host> -> rustc 1 <host>
+        [build] rustc 0 <host> -> clippy-driver 1 <host>
+        [build] rustc 0 <host> -> cargo-clippy 1 <host>
+        [clippy] rustc 1 <host> -> std 1 <host>
+        [clippy] rustc 0 <host> -> rustc 1 <host>
+        [clippy] rustc 0 <host> -> rustc_codegen_gcc 1 <host>
+        ");
+    }
+
+    #[test]
+    fn clippy_ci_stage_2() {
+        let ctx = TestCtx::new();
+        insta::assert_snapshot!(
+            ctx.config("clippy")
+                .path("ci")
+                .stage(2)
+                .render_steps(), @r"
+        [build] llvm <host>
+        [build] rustc 0 <host> -> rustc 1 <host>
+        [build] rustc 1 <host> -> std 1 <host>
+        [build] rustc 0 <host> -> clippy-driver 1 <host>
+        [build] rustc 0 <host> -> cargo-clippy 1 <host>
+        [clippy] rustc 1 <host> -> bootstrap 2 <host>
+        [build] rustc 1 <host> -> rustc 2 <host>
```

Comment:
^ this

Reply:
Hmm, the log shows `[build]`, I see xD I'll check.